backend/src/core/middlewares/file-attachment.middleware.ts

```diff
@@ -48,7 +51,36 @@ function preprocessPotentialIncomingFile(
   next()
 }
 
+// allow attachment only if (1) the user is not using default email address; (2) user's email has been added to the email_from_address table
+// attachment is a high-cost and high-risk feature that we want to limit to users who have been vetted
+async function checkAttachmentPermission(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  if (req.files?.attachments) {
+    const { from } = req.body
+    if (isDefaultFromAddress(from)) {
+      res.status(403).json({
+        message: 'Attachments are not allowed for default email address',
+      })
+      return
+    }
+    const { fromAddress } = parseFromAddress(from)
+    const exists = await CustomDomainService.existsFromAddress(fromAddress)
```

Comment:
This should stop them from even sending the email no?

Reply:
oh I was thinking that is the intended behavior (unless you mean we should just remove the attachment and still send the email?) I think that's a bit trickier since, if the user isn't testing properly, things might fail silently (they send an email talking about some non-existent attachment)

not sure if I'm misunderstanding sth

Reply:
discussed, to simplify requirements to forbidding attachments from default Postman from address

otherwise, the extra check is a bit confusing (considering we alr do the same check later on)